flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/BufferReaderWriterUtil.java

```diff
@@ -241,6 +252,8 @@ static Buffer bufferFromMemorySegment(
 			buffer.tagAsEvent();
 		}
 
+		buffer.setCompressed(isCompressed);
```

Comment:
Although we can not make the `isCompressed` field as final inside `NetworkBuffer`, but we could still pass it along with constructor in this case.